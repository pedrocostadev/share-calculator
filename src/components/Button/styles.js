import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    height: '100%',
  },
  button: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: '100%',
    width: '100%',
  },
  text: {
    fontSize: 24,
    color: 'white',
  },
});

export default styles;
