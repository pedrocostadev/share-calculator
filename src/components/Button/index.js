import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, ScrollView, Text } from 'react-native';
import styles from './styles';

const Button = ({ title, onPress, style }) => (
  <ScrollView contentContainerStyle={{ ...styles.container, ...style }}>
    <TouchableOpacity style={styles.button} onPress={() => onPress(title)}>
      <Text style={styles.text}>{title}</Text>
    </TouchableOpacity>
  </ScrollView>
);

Button.propTypes = {
  title: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  style: PropTypes.object,
};

export default Button;
