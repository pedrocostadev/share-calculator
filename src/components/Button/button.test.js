import React from 'react';
import { render, cleanup, NativeTestEvent, fireEvent } from '@testing-library/react-native';

import Button from './';

describe('<Button />', () => {
  const mockedProps = {
    title: 'Button example',
    onPress: jest.fn(),
  };

  afterEach(cleanup);

  test('should render the title', async () => {
    const { queryAllByText, container } = render(<Button {...mockedProps} />);
    const ocurrences = await queryAllByText(mockedProps.title);
    expect(ocurrences).toHaveLength(1);
    expect(container).toMatchSnapshot();
  });

  test('should call onPress', async () => {
    const { getByText } = render(<Button {...mockedProps} />);
    fireEvent(getByText(mockedProps.title), new NativeTestEvent('press'));
    expect(mockedProps.onPress).toHaveBeenCalledTimes(1);
  });
});
