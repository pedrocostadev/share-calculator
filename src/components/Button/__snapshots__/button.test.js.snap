// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the title 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollView
    contentContainerStyle={
      Object {
        "height": "100%",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 24,
          }
        }
      >
        Button example
      </Text>
    </TouchableOpacity>
  </ScrollView>
</View>
`;
